src/Microsoft.Data/PrimitiveDataFrameColumnContainer.cs

```diff
@@ -0,0 +1,158 @@
+        public PrimitiveDataFrameColumnContainer(T[] values)
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            long ii = 0;
+            long length = values.LongLength;
+            DataFrameBuffer<T> curBuffer;
+            if (Buffers.Count == 0)
+            {
+                curBuffer = new DataFrameBuffer<T>();
+                Buffers.Add(curBuffer);
+            }
+            else
+            {
+                curBuffer = Buffers[Buffers.Count - 1];
+            }
+            for (; ii < length; ii++)
+            {
+                if (curBuffer.Length == curBuffer.MaxCapacity)
+                {
+                    curBuffer = new DataFrameBuffer<T>();
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(values[ii]);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer(IEnumerable<T> values)
+        {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (Buffers.Count == 0)
+            {
+                Buffers.Add(new DataFrameBuffer<T>());
+            }
+            var curBuffer = Buffers[Buffers.Count - 1];
+            foreach (var value in values)
+            {
+                if (curBuffer.Length == curBuffer.MaxCapacity)
+                {
+                    curBuffer = new DataFrameBuffer<T>();
+                    Buffers.Add(curBuffer);
+                }
+                curBuffer.Append(value);
+                Length++;
+            }
+        }
+        public PrimitiveDataFrameColumnContainer() { }
+        public long Length;
+        //TODO:
+        public long NullCount => throw new NotImplementedException();
+        private int GetArrayContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentException($"Index {rowIndex} cannot be greater than the Column's Length {Length}");
+            }
+            int curArrayIndex = 0;
+            int numBuffers = Buffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > Buffers[curArrayIndex].Length)
+            {
+                rowIndex -= Buffers[curArrayIndex].Length;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public IList<T> this[long startIndex, int length]
+        {
+            get
+            {
+                var ret = new List<T>();
+                long endIndex = startIndex + length;
+                int arrayIndex = GetArrayContainingRowIndex(ref startIndex);
+                bool temp = Buffers[arrayIndex][(int)startIndex, length, ret];
+                while (ret.Count < length)
+                {
+                    long nextRowIndex = startIndex + ret.Count + 1;
+                    arrayIndex++;
+                    temp = Buffers[arrayIndex][(int)nextRowIndex, length - ret.Count, ret];
```

Comment:
Even though this is internal, this API usage/pattern is a little unusual:

```[(int)nextRowIndex, length - ret.Count, ret];```

Specifically, passing in an object `ret` to get filled up. It feels like it should just be a standard method call.

Reply:
Right. I don't like that I have to do this either. Basically, this is a slice of length "length" that can begin in the middle of some buffer and spill over into the next buffer. I was trying to get by with passing just 1 return list "ret" to both calls to avoid creating more objects. That's why the return type of the indexer is a bool. Ideally, if there was API that made a List from a Span efficiently, or that allowed me to append to a List from a Span, I would've used that. Unless there's a less ugly way to do this? Definitely open to suggestions

Reply:
I think we can just make this a method and not use an indexer. Honestly, I don't think I've ever seen someone use a C# indexer this way.